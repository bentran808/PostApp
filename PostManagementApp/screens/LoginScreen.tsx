import React, {useState} from 'react';
import {Modal, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import FormButton from '../components/FormButton';
import FormInput from '../components/FormInput';
import Ionicons from 'react-native-vector-icons/Ionicons';
// import axios from 'axios';

type LoginNavigationProp = NativeStackNavigationProp<
    RootStackParamList,
    'Login'
>;
interface LoginProps {
    navigation: LoginNavigationProp;
}

const LoginScreen = ({navigation}: LoginProps) => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [error, setError] = useState('');
    const disableButton = !email || !password;

    const login = async () => {
        // await axios({
        //     method: 'post',
        //     url: 'https://192.168.56.1:3000/auth/login',
        //     data: `email=${email}&password=${password}`,
        //     headers: {
        //         'Content-Type': 'application/x-www-form-urlencoded',
        //         Accept: 'application/json'
        //     }
        // })
        //     .then(function (res) {
        //         console.log(res);
        //         navigation.navigate('Home');
        //     })
        //     .catch(function (err) {
        //         console.log(error);
        //         setError(err);
        //     });
        navigation.navigate('Home');
    };

    return (
        <View style={styles.container}>
            <Ionicons name="logo-react" size={150} color="#0ff" />
            <FormInput
                iconType="user"
                value={email}
                onChangeText={userEmail => setEmail(userEmail)}
                placeholderText="Email"
                keyboardType="email-address"
                autoCorrect={false}
                autoCapitalize="none"
            />
            <FormInput
                iconType="lock"
                value={password}
                onChangeText={userPassword => setPassword(userPassword)}
                placeholderText="Password"
                secureTextEntry={true}
            />
            <Modal
                animationType="slide"
                transparent={true}
                visible={!!error}
                onRequestClose={() => {
                    setError('');
                }}>
                <View style={styles.centeredView}>
                    <View style={styles.modalView}>
                        <Text style={styles.modalTitle}>
                            Something is wrong.
                        </Text>
                        <Text style={styles.modalText}>
                            Please re-enter your email and password.
                        </Text>
                        <TouchableOpacity
                            style={[styles.button, styles.buttonClose]}
                            onPress={() => setError('')}>
                            <Text style={styles.textStyle}>Try Again</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </Modal>
            <FormButton
                title="Login"
                disabled={disableButton}
                color={disableButton ? '#ccc' : '#fff'}
                onPress={login}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        flex: 1,
        justifyContent: 'center',
        padding: 20
    },
    centeredView: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 22
    },
    modalView: {
        margin: 20,
        backgroundColor: '#fff',
        borderRadius: 5,
        padding: 10,
        alignItems: 'center',
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 4,
        elevation: 5,
        maxWidth: 300
    },
    button: {
        borderRadius: 8,
        paddingVertical: 10,
        paddingHorizontal: 20,
        elevation: 2
    },
    buttonClose: {
        backgroundColor: '#2196F3'
    },
    textStyle: {
        color: '#fff',
        fontWeight: 'bold'
    },
    modalTitle: {
        fontWeight: 'bold'
    },
    modalText: {
        marginBottom: 15
    }
});
export default LoginScreen;
