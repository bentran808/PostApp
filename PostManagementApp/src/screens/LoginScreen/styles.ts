import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  inner: {
    padding: 24,
    flex: 1,
    justifyContent: 'center'
  },
  alignCenter: {
    alignItems: 'center',
    marginTop: 20
  },
  textCenter: {
    textAlign: 'center'
  }
});
