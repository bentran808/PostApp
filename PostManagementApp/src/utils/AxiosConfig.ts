import axios from 'axios';

export const axiosInstance = axios.create({
    baseURL: 'http://192.168.56.1:3000/',
    headers: {
        'Content-Type': 'application/json'
    }
});
