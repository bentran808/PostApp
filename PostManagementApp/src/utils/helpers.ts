import AsyncStorage from '@react-native-async-storage/async-storage';

const sortDesc = (arr: Post[] = []) => {
  return arr.sort((a, b) => ((a.id || 0) < (b.id || 0) ? 1 : -1));
};

const formatPrice = (price: Number) => {
  return price.toLocaleString('it-IT', {
    style: 'currency',
    currency: 'VND'
  });
};

const getApprovedPosts = (arr: Post[] = []) => arr.filter((item) => !item.pending);

const setAccessToken = async (access_token: string) =>
  await AsyncStorage.setItem('access_token', access_token);

const getAccessToken = async () => await AsyncStorage.getItem('access_token');

const removeAccessToken = async () => await AsyncStorage.removeItem('access_token');

export {
  sortDesc,
  formatPrice,
  getApprovedPosts,
  setAccessToken,
  getAccessToken,
  removeAccessToken
};
