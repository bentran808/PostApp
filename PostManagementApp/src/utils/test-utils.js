import React from 'react';
import { Provider } from 'react-redux';
import { Provider as PaperProvider } from 'react-native-paper';
import { render } from '@testing-library/react-native';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import authReducer from 'redux/slices/authSlice';
import postReducer from 'redux/slices/postSlices';

const sagaMiddleware = createSagaMiddleware();
const AllTheProviders = ({ children }) => {
  const store = configureStore({
    reducer: {
      auth: authReducer,
      post: postReducer
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false
      }).concat(sagaMiddleware)
  });
  return (
    <Provider store={store}>
      <PaperProvider>{children}</PaperProvider>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
