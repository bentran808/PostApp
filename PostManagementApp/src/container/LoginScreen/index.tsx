import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import React, { useEffect, useState } from 'react';
import { object, string } from 'yup';
import { useAppDispatch, useAppSelector } from '../../hooks';
import { authActions } from '../../redux/slices';
import LoginScreen from '../../screens/LoginScreen';

type LoginNavigationProp = NativeStackNavigationProp<RootStackParamList, 'Home'>;
interface LoginProps {
    navigation: LoginNavigationProp;
}

export interface ErrorState {
    email: string;
    password: string;
    errorMessage: string;
}

const loginSchema = object().shape({
    email: string().email(),
    password: string().min(5)
});

const LoginScreenContainer = ({ navigation }: LoginProps) => {
    const [email, setEmail] = useState<string>('');
    const [password, setPassword] = useState<string>('');
    const [hidden, setHidden] = useState<boolean>(true);
    const [error, setError] = useState<ErrorState>({
        email: '',
        password: '',
        errorMessage: ''
    });
    const dispatch = useAppDispatch();
    const isLoggedIn = useAppSelector((state) => state.auth.isLoggedIn);
    const errorMessage = useAppSelector((state) => state.auth.errorMessage);
    const hasError = Object.values(error).some((item) => item !== '');

    useEffect(() => {
        if (isLoggedIn) {
            navigation.navigate('Home');
            setEmail('');
            setPassword('');
        }
    }, [isLoggedIn, navigation]);

    const handleChangeEmail = async (userEmail: string) => {
        const isValid = loginSchema.isValidSync({ email: userEmail });
        setEmail(userEmail);
        loginSchema
            .validate({ email: userEmail })
            .catch((err) => setError({ ...error, email: err.errors[0] }));
        if (isValid) {
            setError({ ...error, email: '' });
        }
    };

    const handleChangePassword = (userPassword: string) => {
        const isValid = loginSchema.isValidSync({ password: userPassword });
        setPassword(userPassword);
        loginSchema
            .validate({ password: userPassword })
            .catch((err) => setError({ ...error, password: err.errors[0] }));
        if (isValid) {
            setError({ ...error, password: '' });
        }
    };

    const handleHiddenPassword = () => setHidden(!hidden);

    const handleLogin = () => {
        dispatch(authActions.login(`email=${email}&password=${password}`));
        setError({ ...error, errorMessage });
    };

    return (
        <LoginScreen
            email={email}
            password={password}
            hidden={hidden}
            error={error}
            hasError={hasError}
            onSetError={setError}
            onChangeEmail={handleChangeEmail}
            onChangePassword={handleChangePassword}
            onHiddenPassword={handleHiddenPassword}
            onLogin={handleLogin}
        />
    );
};

export default LoginScreenContainer;
