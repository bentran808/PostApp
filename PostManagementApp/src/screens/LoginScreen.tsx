import React, {useContext, useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {Button, Dialog, Paragraph, Portal, TextInput} from 'react-native-paper';

// Utilities
import {axiosInstance} from '../utils/AxiosConfig';
import {AppContext} from '../navigation/AppContext';
import {windowWidth} from '../utils/Dimensions';
import {colors} from '../constants/colors';

type LoginNavigationProp = NativeStackNavigationProp<
    RootStackParamList,
    'Login'
>;
interface LoginProps {
    navigation: LoginNavigationProp;
}

const LoginScreen = ({navigation}: LoginProps) => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [hidden, setHidden] = useState(true);
    const [error, setError] = useState<unknown>();
    const {setUser} = useContext(AppContext);
    const disableButton = !email || !password;

    const login = async () => {
        try {
            const response = await axiosInstance.post(
                'auth/login',
                `email=${email}&password=${password}`,
                {
                    headers: {
                        'Content-Type': 'application/x-www-form-urlencoded',
                        Accept: 'application/json'
                    }
                }
            );
            if (response.status === 200) {
                setEmail('');
                setPassword('');
                console.log(response);
                setUser(response.data);
                navigation.navigate('Home');
            } else {
                throw new Error('An error has occurred');
            }
        } catch (err) {
            setError(err);
            console.log(err);
        }
    };

    return (
        <>
            <View style={styles.container}>
                <Ionicons
                    name="logo-react"
                    size={150}
                    color="#0ff"
                    style={styles.textCenter}
                />
                <TextInput
                    mode="outlined"
                    activeOutlineColor={colors.royalBlue}
                    label="Email"
                    value={email}
                    onChangeText={userEmail => setEmail(userEmail)}
                    keyboardType="email-address"
                    autoCorrect={false}
                    autoCapitalize="none"
                    style={styles.marginBottom}
                />
                <TextInput
                    mode="outlined"
                    activeOutlineColor={colors.royalBlue}
                    label="Password"
                    secureTextEntry={hidden}
                    right={
                        <TextInput.Icon
                            name={hidden ? 'eye' : 'eye-off'}
                            onPress={() => setHidden(!hidden)}
                        />
                    }
                    value={password}
                    onChangeText={userPassword => setPassword(userPassword)}
                    style={styles.marginBottom}
                />
                <View style={styles.alignCenter}>
                    <Button
                        mode="contained"
                        color={colors.royalBlue}
                        disabled={disableButton}
                        uppercase={false}
                        style={{
                            width: windowWidth / 2
                        }}
                        onPress={login}>
                        Login
                    </Button>
                </View>
                <Portal>
                    <Dialog visible={!!error} onDismiss={() => setError('')}>
                        <Dialog.Title>Something is wrong.</Dialog.Title>
                        <Dialog.Content>
                            <Paragraph>
                                Please re-enter your email and password.
                            </Paragraph>
                        </Dialog.Content>
                        <Dialog.Actions>
                            <Button
                                color={colors.royalBlue}
                                uppercase={false}
                                onPress={() => setError('')}>
                                Try Again
                            </Button>
                        </Dialog.Actions>
                    </Dialog>
                </Portal>
            </View>
        </>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        padding: 20
    },
    alignCenter: {
        alignItems: 'center'
    },
    textCenter: {
        textAlign: 'center'
    },
    marginBottom: {
        marginBottom: 10
    }
});
export default LoginScreen;
