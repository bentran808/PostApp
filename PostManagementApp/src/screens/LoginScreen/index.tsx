import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { Screens } from 'constants/screens';
import { useAppDispatch } from 'hooks';
import React, { useEffect, useMemo, useState } from 'react';
import {
  Alert,
  Keyboard,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
  View
} from 'react-native';
import { Button, HelperText, TextInput } from 'react-native-paper';
import SplashScreen from 'react-native-splash-screen';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { colors } from 'theme/Colors';
import { windowWidth } from 'utils/Dimensions';
import { object, string } from 'yup';
import { authActions } from '../../redux/slices';
import { styles } from './styles';

type LoginNavigationProp = NativeStackNavigationProp<RootStackParamList, 'Home'>;
interface LoginProps {
  navigation: LoginNavigationProp;
}

const loginSchema = object().shape({
  email: string().email(),
  password: string().min(5)
});

const validate = (key: string, value: string) => {
  try {
    loginSchema.validateSync({ [key]: value });
  } catch (err: any) {
    return err.errors[0];
  }
};

const LoginScreen = ({ navigation }: LoginProps) => {
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [hidden, setHidden] = useState<boolean>(true);
  const dispatch = useAppDispatch();
  const disableButton = !email || !password;
  const { error } = useMemo(() => {
    const isValid = loginSchema.isValidSync({ email, password });
    if (isValid) {
      return {
        error: {
          email: '',
          password: ''
        }
      };
    } else {
      return {
        error: {
          email: email ? validate('email', email) : '',
          password: password ? validate('password', password) : ''
        }
      };
    }
  }, [email, password]);

  useEffect(() => {
    SplashScreen.hide();
  });

  const handleChangeEmail = (userEmail: string) => {
    setEmail(userEmail);
  };

  const handleChangePassword = (userPassword: string) => {
    setPassword(userPassword);
  };

  const handleHiddenPassword = () => setHidden(!hidden);

  const handleLogin = () => {
    dispatch(
      authActions.login({
        params: `email=${email}&password=${password}`,
        callback: (data, err) => {
          if (data) {
            navigation.navigate(Screens.HOME.name as 'Home');
            setEmail('');
            setPassword('');
            setHidden(true);
          }

          if (err) {
            Alert.alert(err.response.data.message);
          }
        }
      })
    );
  };

  return (
    <KeyboardAvoidingView behavior="height" style={styles.container}>
      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
        <View style={styles.inner}>
          <Ionicons name="logo-react" size={150} color="#0ff" style={styles.textCenter} />
          <TextInput
            mode="outlined"
            activeOutlineColor={colors.royalBlue}
            label="Email"
            value={email}
            onChangeText={handleChangeEmail}
            keyboardType="email-address"
            autoFocus
            autoCorrect={false}
            autoCapitalize="none"
            error={!!error.email}
            testID="emailInput"
          />
          <HelperText type="error" visible={!!error.email} testID="emailError">
            {error.email}
          </HelperText>
          <TextInput
            mode="outlined"
            activeOutlineColor={colors.royalBlue}
            label="Password"
            secureTextEntry={hidden}
            right={
              <TextInput.Icon
                name={hidden ? 'eye' : 'eye-off'}
                onPress={handleHiddenPassword}
                testID="icon"
              />
            }
            value={password}
            onChangeText={handleChangePassword}
            error={!!error.password}
            testID="passwordInput"
          />
          <HelperText type="error" visible={!!error.password} testID="passwordError">
            {error.password}
          </HelperText>
          <View style={styles.alignCenter}>
            <Button
              mode="contained"
              color={colors.royalBlue}
              disabled={disableButton}
              uppercase={false}
              style={{
                width: windowWidth / 2
              }}
              onPress={handleLogin}
              testID="loginButton"
            >
              Login
            </Button>
          </View>
        </View>
      </TouchableWithoutFeedback>
    </KeyboardAvoidingView>
  );
};

export default LoginScreen;
