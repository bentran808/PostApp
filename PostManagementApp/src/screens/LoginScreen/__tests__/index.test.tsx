import React from 'react';
import LoginScreen from '..';
import * as customHooks from '../../../hooks';
import { fireEvent, render } from '../../../utils/test-utils';

jest.mock('../../../hooks');

describe('Test Login Screen', () => {
  const dispatch = jest.fn();
  const spy = jest.spyOn(customHooks, 'useAppDispatch');
  spy.mockReturnValue(dispatch);
  let navigation: any;
  beforeEach(() => {
    navigation = {
      navigate: jest.fn()
    };
  });

  test('should render correctly', () => {
    const { toJSON } = render(<LoginScreen navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('should show email error message', () => {
    const { getByTestId } = render(<LoginScreen navigation={navigation} />);
    const input = getByTestId('emailInput');
    fireEvent.changeText(input, 'asd');
    const message = getByTestId('emailError');
    expect(message).toBeDefined();
    expect(message.children).toEqual(['email must be a valid email']);
  });

  test('should show password error message', () => {
    const { getByTestId } = render(<LoginScreen navigation={navigation} />);
    const input = getByTestId('passwordInput');
    fireEvent.changeText(input, '1');
    const message = getByTestId('passwordError');
    expect(message).toBeDefined();
    expect(message.children).toEqual(['password must be at least 5 characters']);
  });

  test('should change hidden icon', () => {
    const { getByTestId } = render(<LoginScreen navigation={navigation} />);
    const icon = getByTestId('icon');
    fireEvent.press(icon);
    expect(getByTestId('passwordInput').props.secureTextEntry).toBeFalsy();
  });

  test('should call function handleLogin', () => {
    const { getByTestId } = render(<LoginScreen navigation={navigation} />);
    const email = getByTestId('emailInput');
    fireEvent.changeText(email, 'admin@admin.com');
    const password = getByTestId('passwordInput');
    fireEvent.changeText(password, 'admin');
    const button = getByTestId('loginButton');
    fireEvent.press(button);

    expect(dispatch).toHaveBeenCalled();
  });
});
