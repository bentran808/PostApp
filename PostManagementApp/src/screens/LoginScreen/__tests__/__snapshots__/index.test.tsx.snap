// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Screen should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "padding": 24,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#0ff",
              "fontSize": 150,
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïŽ¾
      </Text>
      <View
        style={Object {}}
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#2e64e5",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Email
              </Text>
            </View>
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              autoFocus={true}
              editable={true}
              keyboardType="email-address"
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              selectionColor="#2e64e5"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="emailInput"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <Text
        collapsable={false}
        onLayout={[Function]}
        style={
          Object {
            "color": "#B00020",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "400",
            "opacity": 0,
            "paddingHorizontal": 12,
            "paddingVertical": 4,
            "textAlign": "left",
            "transform": Array [],
            "writingDirection": "ltr",
          }
        }
        testID="emailError"
      >
        
      </Text>
      <View
        style={Object {}}
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "minHeight": 64,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "bottom": 4,
                    "left": 10,
                    "maxHeight": 2,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                    "width": 12,
                  }
                }
              />
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 18,
                    "maxWidth": -28,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#2e64e5",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                Password
              </Text>
            </View>
            <TextInput
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              secureTextEntry={true}
              selectionColor="#2e64e5"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                    "paddingHorizontal": 14,
                    "zIndex": 1,
                  },
                  Object {
                    "height": 56,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Object {
                    "marginRight": 44,
                    "paddingRight": 0,
                  },
                ]
              }
              testID="passwordInput"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 24,
                },
                Object {
                  "right": 12,
                  "top": 23,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
              testID="icon"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ó°ˆˆ
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        collapsable={false}
        onLayout={[Function]}
        style={
          Object {
            "color": "#B00020",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "400",
            "opacity": 0,
            "paddingHorizontal": 12,
            "paddingVertical": 4,
            "textAlign": "left",
            "transform": Array [],
            "writingDirection": "ltr",
          }
        }
        testID="passwordError"
      >
        
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 20,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 0,
              "minWidth": 64,
              "width": 375,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
            testID="loginButton"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(0, 0, 0, 0.32)",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Login
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
